var/spack/repos/builtin/packages/gptune/package.py

```diff
@@ -199,10 +199,7 @@ def cache_test_sources(self):
     def setup_run_environment(self, env):
         env.set("GPTUNE_INSTALL_PATH", python_platlib)
 
-    bash = which("bash")
-    cp = which("cp")
-    git = which("git")
-    rm = which("rm")
+    cmd = {"bash": which("bash"), "cp": which("cp"), "git": which("git"), "rm": which("rm")}
```

Comment:
Running `which` here means that you will get versions of these commands found when the package is imported, not versions that exist when the test environment set up (at which point test deps are activated).  In general I think it's bad practice to call `which()` outside the test environment itself.